CMakeLists.txt

```diff
@@ -60,49 +67,71 @@ else()
     set (CLANG_BIN ${CLANG_INSTALL_PREFIX}/bin/clang++)
 endif()
 
-execute_process(COMMAND ${CLANG_BIN} -print-resource-dir
-    RESULT_VARIABLE CLANG_RESOURCE_DIR_RESULT
-    OUTPUT_VARIABLE CLANG_RESOURCE_DIR
-    ERROR_VARIABLE  CLANG_RESOURCE_DIR_ERROR
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-if(CLANG_RESOURCE_DIR_RESULT)
-    message(FATAL_ERROR "Could not locate the resource dir for clang we are going to link against")
-    message(FATAL_ERROR ${CLANG_RESOURCE_DIR_ERROR})
-endif()
+# execute_process(COMMAND ${CLANG_BIN} -print-resource-dir
+#     RESULT_VARIABLE CLANG_RESOURCE_DIR_RESULT
+#     OUTPUT_VARIABLE CLANG_RESOURCE_DIR
+#     ERROR_VARIABLE  CLANG_RESOURCE_DIR_ERROR
+#     OUTPUT_STRIP_TRAILING_WHITESPACE)
+# 
+# if(CLANG_RESOURCE_DIR_RESULT)
+#     message(FATAL_ERROR "Could not locate the resource dir for clang we are going to link against")
+#     message(FATAL_ERROR ${CLANG_RESOURCE_DIR_ERROR})
+# endif()
 
 message(STATUS "Using the following resource dir for clang: ${CLANG_RESOURCE_DIR}")
 
 file(GLOB EMITTER_FILES CONFIGURE_DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/emitters/*.cpp)
 file(GLOB MATCHER_FILES CONFIGURE_DEPENDS ${PROJECT_SOURCE_DIR}/matchers/*.cpp)
 file(GLOB SRC_FILES CONFIGURE_DEPENDS ${PROJECT_SOURCE_DIR}/sources/*.cpp)
-
 file(GLOB YAML_CPP_SRC_FILES CONFIGURE_DEPENDS ${PROJECT_SOURCE_DIR}/submodules/yaml-cpp/src/*.cpp)
 
 #TODO: use target_source
-add_executable(hyde  ${EMITTER_FILES} ${MATCHER_FILES} ${SRC_FILES} ${YAML_CPP_SRC_FILES})
+add_executable(hyde
+    ${EMITTER_FILES}
+    ${MATCHER_FILES}
+    ${SRC_FILES}
+    ${YAML_CPP_SRC_FILES}
+)
+
+target_compile_definitions(hyde
+    PRIVATE
+        CLANG_RESOURCE_DIR="${CLANG_RESOURCE_DIR}"
+        LLVM_INCLUDE_DIR="${llvm_INCLUDE_DIR}"
+)
+
+target_include_directories(hyde
+    PUBLIC
+        ${Boost_INCLUDE_DIRS}
+        ${CLANG_INCLUDE_DIRS}
```

Comment:
`CLANG_INCLUDE_DIRS` will not be set when building llvm as a subdirectory. Normally you would only need to link against the appropriate targets to get the corresponding include directories but the llvm targets are not quite correct in this regard, so you may have to add include directories manually, something like
```
target_include_directories(hyde
    PUBLIC
        ${llvm_SOURCE_DIR}/clang/include
        ${llvm_BINARY_DIR}/tools/clang/include
        ${llvm_SOURCE_DIR}/llvm/include
        ${llvm_BINARY_DIR}/include
)
```